packages/contentstack-export-to-csv/src/commands/cm/export-to-csv.js

```diff
@@ -292,9 +260,149 @@ class ExportToCsvCommand extends Command {
+        name: stackName || managementTokenAlias,
+        apiKey: listOfTokens[managementTokenAlias].apiKey,
+        token: listOfTokens[managementTokenAlias].token,
+      };
+    } else if (managementTokenAlias) {
+      this.error('Provided management token alias not found in your config.!');
+    }
+    return {
+      apiClient,
+      stackDetails,
+    };
+  }
+
+  /**
+   * fetch stack details on basis of the selected org and stack
+   * @param {*} managementAPIClient
+   * @param {string} stackAPIKey
+   * @param {string} org
+   * @returns
+   */
+  async getStackDetails(managementAPIClient, stackAPIKey, org) {
+    let organization, stackDetails;
+
+    if (!isAuthenticated()) {
+      this.error(config.CLI_EXPORT_CSV_ENTRIES_ERROR, {
+        exit: 2,
+        suggestions: ['https://www.contentstack.com/docs/developers/cli/authentication/'],
+      });
+    }
+
+    if (org) {
+      organization = { uid: org };
+    } else {
+      organization = await util.chooseOrganization(managementAPIClient); // prompt for organization
+    }
+    if (!stackAPIKey) {
+      stackDetails = await util.chooseStack(managementAPIClient, organization.uid); // prompt for stack
+    } else {
+      stackDetails = await util.chooseStack(managementAPIClient, organization.uid, stackAPIKey);
+    }
+    return stackDetails;
+  }
+
+  /**
+   * Create a taxonomies csv file for stack and a terms csv file for associated taxonomies
+   * @param {string} stackName
+   * @param {object} stack
+   * @param {string} taxUID
+   */
+  async createTaxonomyAndTermCsvFile(stackAPIClient, stackName, stack, taxUID) {
+    const payload = {
+      stackAPIClient,
+      type: '',
+      limit: config.limit || 100
+    };
+    //check whether the taxonomy is valid or not
+    let taxonomies = [];
+    if (taxUID) {
+      payload['taxonomyUID'] = taxUID;
+      const taxonomy = await util.getTaxonomy(payload);
+      taxonomies.push(taxonomy);
+    } else {
+      taxonomies = await util.getAllTaxonomies(payload);
+    }
+
+    const formattedTaxonomiesData = util.formatTaxonomiesData(taxonomies);
+    if (formattedTaxonomiesData?.length) {
+      const fileName = `${stackName ? stackName : stack.name}_taxonomies.csv`;
+      util.write(this, formattedTaxonomiesData, fileName, 'taxonomies');
+    } else {
+      cliux.print('info: No taxonomies found! Please provide a valid stack.', { color: 'blue' });
+    }
+
+    for (let index = 0; index < taxonomies?.length; index++) {
+      const taxonomy = taxonomies[index];
+      const taxonomyUID = taxonomy?.uid;
+      if (taxonomyUID) {
+        payload['taxonomyUID'] = taxonomyUID;
+        const terms = await util.getAllTermsOfTaxonomy(payload);
+        const formattedTermsData = util.formatTermsOfTaxonomyData(terms, taxonomyUID);
+        const taxonomyName = taxonomy?.name ? taxonomy.name : '';
+        const termFileName = `${stackName ? stackName : stack.name}_${taxonomyName}_${taxonomyUID}_terms.csv`;
```

Comment:
Sometimes this syntax throws a (`Nullish coalescing operator`) lint issue and instead, it's suggested to use like following.

`taxonomy?.name ?? ''`
`stackName ?? stack.name`

Reply:
Done #1130